File: app/src/main/java/com/possible/architecturesample/data/network/requests/BaseRequest.java
package com.possible.architecturesample.data.network.requests;

import java.util.HashMap;
import java.util.Map;

public class BaseRequest {

    protected String createUrl(String url, HashMap<String, String> map) {
        if (map != null) {
            for (Map.Entry<String, String> entry : map.entrySet()) {
                url = url.replace(entry.getKey(), entry.getValue());
            }
        }
        return url;
    }
}

File: app/src/main/java/com/possible/architecturesample/data/network/requests/BaseRequest.kt
package com.possible.architecturesample.data.network.requests

import java.util.HashMap

open class BaseRequest {

    protected fun createUrl(url: String, map: HashMap<String, String>?): String {
        var newUrl = url
        if (map != null) {
            for ((key, value) in map) {
                newUrl = newUrl.replace(key, value)
            }
        }
        return newUrl
    }
}
